applications/ConstitutiveLawsApplication/custom_constitutive/auxiliary_files/hcf_data_container.h

```diff
@@ -0,0 +1,313 @@
+        rFatigueVariables.PreviousMaxStress = mPreviousMaxStress;
+        rFatigueVariables.PreviousMinStress = mPreviousMinStress;
+        rFatigueVariables.WohlerStress = mWohlerStress;
+        rFatigueVariables.NewCycle = false;
+        rFatigueVariables.Sth = mThresholdStress;
+        rFatigueVariables.CyclesToFailure = mCyclesToFailure;
+    }
+
+    /**
+     * @brief This method updates all member variables
+     */
+    void UpdateFatigueVariables(HCFDataContainer::FatigueVariables &rFatigueVariables)
+    {
+        mMaxStress = rFatigueVariables.MaxStress;
+        mMinStress = rFatigueVariables.MinStress;
+        mMaxDetected = rFatigueVariables.MaxIndicator;
+        mMinDetected = rFatigueVariables.MinIndicator;
+        mNumberOfCyclesGlobal = rFatigueVariables.GlobalNumberOfCycles;
+        mNumberOfCyclesLocal = rFatigueVariables.LocalNumberOfCycles;
+        mNewCycleIndicator = rFatigueVariables.NewCycle;
+        mFatigueReductionParameter = rFatigueVariables.B0;
+        mPreviousMaxStress = rFatigueVariables.PreviousMaxStress;
+        mPreviousMinStress = rFatigueVariables.PreviousMinStress;
+        mPreviousStresses = rFatigueVariables.PreviousStresses;
+        mFatigueReductionFactor = rFatigueVariables.FatigueReductionFactor;
+        mWohlerStress = rFatigueVariables.WohlerStress;
+        mThresholdStress = rFatigueVariables.Sth;
+        mReversionFactorRelativeError = rFatigueVariables.ReversionFactorRelativeError;
+        mMaxStressRelativeError = rFatigueVariables.MaxStressRelativeError;
+    }
+
+    /**
+     * @brief This method computes fatigue-related quantities
+     */
+    void FinalizeSolutionStep(HCFDataContainer::FatigueVariables &rFatigueVariables,
+                            const Properties& rMaterialProperties,
+                            const ProcessInfo& rCurrentProcessInfo,
+                            ConstitutiveLaw::StressVectorType stress_vector,
+                            double uniaxial_stress)
+    {
+        double sign_factor = CalculateTensionOrCompressionIdentifier(stress_vector);
+        uniaxial_stress *= sign_factor;
+
+        CalculateSminAndSmax(uniaxial_stress, rFatigueVariables);
+
+        rFatigueVariables.AdnvanceStrategyApplied = rCurrentProcessInfo[ADVANCE_STRATEGY_APPLIED];
+        rFatigueVariables.DamageActivation = rCurrentProcessInfo[DAMAGE_ACTIVATION];
+
+        if (rFatigueVariables.MaxIndicator && rFatigueVariables.MinIndicator) {
+            rFatigueVariables.PreviousReversionFactor = CalculateReversionFactor(rFatigueVariables.PreviousMaxStress, rFatigueVariables.PreviousMinStress);
+            rFatigueVariables.ReversionFactor = CalculateReversionFactor(rFatigueVariables.MaxStress, rFatigueVariables.MinStress);
+
+            CalculateFatigueParameters(rMaterialProperties, rFatigueVariables);
+
+            double betaf = rMaterialProperties[HIGH_CYCLE_FATIGUE_COEFFICIENTS][4];
+
+            if (std::abs(rFatigueVariables.MinStress) < 0.001) {
```

Comment:
we should set this tolerance as a `static constexpr double tolerance = 1.0e-3;` like we do in the plasticity integrator